Source/IGListAdapter.m

```diff
@@ -534,15 +534,26 @@ - (void)updateBackgroundViewShouldHide:(BOOL)shouldHide {
     if (self.isInUpdateBlock) {
         return; // will be called again when update block completes
     }
-    UIView *backgroundView = [self.dataSource emptyViewForListAdapter:self];
-    // don't do anything if the client is using the same view
-    if (backgroundView != _collectionView.backgroundView) {
-        // collection view will just stack the background views underneath each other if we do not remove the previous
-        // one first. also fine if it is nil
-        [_collectionView.backgroundView removeFromSuperview];
-        _collectionView.backgroundView = backgroundView;
+
+    if (shouldHide) {
+        _emptyListView.hidden = YES;
+    } else {
+        UIView *const newBackgroundView = [self.dataSource emptyViewForListAdapter:self];
+        if (_emptyListView != newBackgroundView) {
+            [_emptyListView removeFromSuperview];
+            _emptyListView = newBackgroundView;
+        }
+
+        if (_emptyListView != nil) {
+            // This makes the compiler gods happy.
+            UIView *_Nonnull const nonullEmptyView = _emptyListView;
```

Comment:
I need to keep the `_Nonnull` or I get a compiler error. Since `emptyViewForListAdapter:` returns `nullable UIView *` it won't let me add it as subview to `_collectionView` without casting it because it is worried the view could be nil.